File: contractstudy/src/test/java/test/contractstudy/collectContracts/cre/JavaCREExtractor/java/testData/MultipleJavaCRE.java
package test.contractstudy.collectContracts.cre.JavaCREExtractor.java.testData;


import android.content.res.Resources;

import java.time.DateTimeException;
import java.util.List;

public class MultipleJavaCRE {

  public void test1(List<String> list) {
    if (list.isEmpty()) {
      throw new IllegalArgumentException("List is empty");
    }
    throw new IllegalArgumentException("This is not a JavaCRE");
  }

  public void test2(boolean flag, List<String> list) {
    if (!flag) {
      throw new Resources.NotFoundException("Flag is false");
    }
  }

  public void test3(List<String> list) {
    if (list.size() < 5) {
      throw new IndexOutOfBoundsException("Smaller than 5");
    }
    throw new DateTimeException("This is not an JavaCRE");
  }
}

File: contractstudy/src/test/java/contractstudy/collectContracts/cre/JavaCREExtractor/kotlin/testData/MultipleJavaCRE.kt
package contractstudy.collectContracts.cre.JavaCREExtractor.kotlin.testData

import android.content.res.Resources
import java.time.DateTimeException

class MultipleJavaCRE {
    fun test1(list: List<String?>) {
        if (!list.isEmpty()) {
            throw IllegalArgumentException("List is empty")
        }
        throw IllegalArgumentException("This is not a JavaCRE")
    }

    fun test2(flag: Boolean, list: List<String?>?) {
        if (!flag) {
            throw Resources.NotFoundException("Flag is false")
        }
    }

    fun test3(list: List<String?>) {
        if (list.size < 5) {
            throw IndexOutOfBoundsException("Smaller than 5")
        }
        throw DateTimeException("This is not an JavaCRE")
    }
}